src/django/api/management/commands/fix_merged_rejections.py

```diff
@@ -0,0 +1,57 @@
+from django.core.management.base import BaseCommand
+
+from api.models import FacilityMatch, HistoricalFacilityMatch
+
+
+class Command(BaseCommand):
+    help = ('Reset incorrectly merged facility list items.')
+
+    def add_arguments(self, parser):
+        parser.add_argument('-d', '--dry-run', action='store_true',
+                            default=False, help='Do not save cahnges')
+
+    def handle(self, *args, **options):
+        dry_run = options['dry_run']
+
+        if dry_run:
+            print('== DRY RUN - SKIPPING ALL SAVES')
+
+        count = 0
+        for hist_match in HistoricalFacilityMatch.objects.filter(
+            status='MERGED'
+        ):
+            prev_record = hist_match.prev_record
+            if prev_record.status in ['REJECTED', 'PENDING']:
+                list_item = prev_record.facility_list_item
+                valid_matches = FacilityMatch \
+                    .objects \
+                    .filter(facility_list_item_id=list_item.id,
+                            status__in=['AUTOMATIC', 'CONFIRMED'])
+                if len(valid_matches) > 1:
+                    print(
+                        'Multiple valid matches found for list item {}'.format(
+                            list_item.id
+                        ))
+                elif len(valid_matches) < 1:
+                    print(
+                        'No valid matches found for list item {}'.format(
+                            list_item.id
+                        ))
+                else:
+                    count = count + 1
+                    valid_match = valid_matches[0]
+                    list_item.facility_id = valid_match.facility_id
+                    if not dry_run:
+                        list_item.save()
+                        print(f'item,save,{list_item.pk}')
+                    else:
+                        print(f'item,dry,{list_item.pk}')
+
+                    hist_match.status = prev_record.status
+                    if not dry_run:
+                        hist_match.save()
+                        print(f'match,save,{hist_match.pk}')
+                    else:
+                        print(f'match,dry,{hist_match.pk}')
```

Comment:
Do we need to, and is it wise to, "rewrite history?" I'm a bit nervous about directly manipulating `HistoricalFacilityMatch` objects since they are just a reflection of the various statues that a `FacilityMatch` has gone through.

Should we instead consider updating the status on the actual `FacilityMatch` and specifying a `changeReason` that explains we are fixing the data?


```suggestion
                    match = FacilityMatch.objects.get(pk=hist_match.id)
                    if not dry_run:
                        match.status = prev_record.status
                        match.changeReason = 'Fixing merge rejection'
                        match.save()
                        print(f'match,save,{match.pk}')
                    else:
                        print(f'match,dry,{match.pk}')
```

Reply:
In line with this, should we add a processing result to the list item with a new action (something like, UNDO_MERGE)?

Reply:
I've pushed a fix up that also adds a processing result to the list item.